base/server/python/pki/server/cli/user.py

```diff
@@ -353,6 +354,81 @@ def execute(self, argv):
             del_see_also=del_see_also)
 
 
+class UserRemoveCLI(pki.cli.CLI):
+
+    def __init__(self, parent):
+        super(UserRemoveCLI, self).__init__(
+            'del',
+            'Remove %s user' % parent.parent.name.upper())
+
+        self.parent = parent
+
+    def print_help(self):
+        print('Usage: pki-server %s-user-del [OPTIONS] <user ID>' % self.parent.parent.name)
+        print()
+        print('  -i, --instance <instance ID>       Instance ID (default: pki-tomcat).')
+        print('  -v, --verbose                      Run in verbose mode.')
+        print('      --debug                        Run in debug mode.')
+        print('      --help                         Show help message.')
+        print()
+
+    def execute(self, argv):
+        try:
+            opts, args = getopt.gnu_getopt(argv, 'i:v', [
```

Comment:
I tried using `argparse` to mimic the current CLIs, but I ended up with a longer code and it has a slightly different look & feel. Here's the code:

```
        parser = argparse.ArgumentParser(
            prog='pki-server ' + self.get_full_name(),
            usage='%(prog)s [OPTIONS] USER_ID',
            description=self.description,
            add_help=False)

        parser.add_argument(
            '--instance',
            help='Instance ID (default: pki-tomcat)',
            default='pki-tomcat')

        parser.add_argument(
            '-v', '--verbose',
            help='Run in debug mode.',
            action='store_true')

        parser.add_argument(
            '--debug',
            help='Run in verbose mode.',
            action='store_true')

        parser.add_argument(
            '--help',
            help='Show help message.',
            action='store_true')

        parser.add_argument(
            'USER_ID',
            nargs='?',
            help='User ID')

        args = parser.parse_args(argv)

        if args.help:
            parser.print_help()
            sys.exit()

        if args.USER_ID is None:
            logger.error('Missing user ID')
            parser.print_help()
            sys.exit(1)

        if args.verbose:
            logging.getLogger().setLevel(logging.INFO)

        if args.debug:
            logging.getLogger().setLevel(logging.DEBUG)

        instance_name = args.instance
        subsystem_name = self.parent.parent.name
        user_id = args.USER_ID
```
So for now I'll just use `getopt` in this PR, but please feel free to try the above code and see if you can improve it.

Reply:
Yeah, that is the trade off with `argparse` unfortunately. What I have done in the past was pull out a function or two for all the arg related things.

It would look something like:

```python
args = setup_args() 

if args.help: 
    parser.print_help() 
    sys.exit() 

if args.USER_ID is None: 
    logger.error('Missing user ID') 
    parser.print_help() 
    sys.exit(1) 

if args.verbose: 
    logging.getLogger().setLevel(logging.INFO) 

if args.debug: 
    logging.getLogger().setLevel(logging.DEBUG) 

instance_name = args.instance 
subsystem_name = self.parent.parent.name 
user_id = args.USER_ID
```